src/System.Windows.Forms/src/System/Windows/Forms/ToolStripRenderer.cs

```
@@ -596,12 +596,19 @@ protected static void ScaleArrowOffsetsIfNeeded(int dpi)
             offset4Y = DpiHelper.LogicalToDeviceUnits(OFFSET_4PIXELS, dpi);
         }
 
-        protected virtual void OnRenderArrow(ToolStripArrowRenderEventArgs e){
+        protected virtual void OnRenderArrow(ToolStripArrowRenderEventArgs e)
+        {
             if (RendererOverride != null)
             {
                 RendererOverride.OnRenderArrow(e);
                 return;
             }
+
+            if (e == null)
+            {
+                throw new ArgumentNullException(nameof(e));
+            }
```

This a somewhat peculiar placement, though it will be a potentially breaking behaviour. I would still argue it isn't expected to `e = null`, and thus it may be justified to fail fast.

I'll have an internal discussion about it, but either way we'll have to update the docs to reflect the new behaviour.

---

> This a somewhat peculiar placement

I wasn't sure if it should be before or after the renderer overide, let me know what you think

---

We had an internal discussion and concluded it would be extremely unlikely an `RendererOverride` would correctly function with `e == null`.
So let's move the check as the first thing.

In general we prefer to perform input arg validation first before any other functionality.

---

Done, thanks